x-pack/plugins/lens/public/indexpattern_datasource/visualize_geo_field_button.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, MouseEvent } from 'react';
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { getUiActions } from '../kibana_services';
+import {
+  visualizeGeoFieldTrigger,
+  VISUALIZE_GEO_FIELD_TRIGGER,
+} from '../../../../../src/plugins/ui_actions/public';
+
+interface Props {
+  indexPatternId: string;
+  fieldName: string;
+}
+
+interface State {
+  hasLoadedHref: boolean;
+  href: string | null;
+}
+
+export class VisualizeGeoFieldButton extends Component<Props, State> {
+  private _isMounted = false;
+  state: State = {
+    hasLoadedHref: false,
+    href: null,
+  };
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.loadHref();
+  }
+
+  async loadHref() {
+    const actions = await getUiActions().getTriggerCompatibleActions(VISUALIZE_GEO_FIELD_TRIGGER, {
+      indexPatternId: this.props.indexPatternId,
+      fieldName: this.props.fieldName,
+    });
+    if (!this._isMounted) {
+      return;
+    }
+
+    const href = actions.length
+      ? await actions[0].getHref?.({
+          indexPatternId: this.props.indexPatternId,
+          fieldName: this.props.fieldName,
+          trigger: visualizeGeoFieldTrigger,
+        })
+      : null;
+
+    this.setState({
+      hasLoadedHref: true,
+      href,
+    });
+  }
+
+  onClick = (event: MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    getUiActions().getTrigger(visualizeGeoFieldTrigger).exec({
```

Comment:
The new pattern is that in `indexpattern_datasource/index.ts` we load the dependencies that we need and pass them into the `getIndexPatternDatasource` function. This button will need to be passed in the uiActions dependency through props from there. Using a React provider is also possible, but Lens has an unusual hierarchy with multiple react roots so this is not something we always choose to do.

If you feel like implementing this change please do, otherwise we may need to take it over.